timelock-server/src/main/java/com/palantir/atlasdb/timelock/lock/AsyncLock.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.lock;
```

Comment:
Should we put all classes in this PR into the `com.palantir.atlasdb.timelock.v2` package to keep the different API versions isolated.

I'm imagining that version X of the service supports version X of the API as standard and *may* include experimental support for version X+1 of the API.  So we'd want all endpoints to include "v2" as part of the path to keep it isolated.